src/TMap.cpp

```diff
@@ -139,6 +139,15 @@ TMap::~TMap()
     }
 }
 
+void TMap::initialiseMembersFromHost()
+{
+    mPlayerRoomColorPrimary = mpHost->mPlayerRoomColorPrimary;
```

Comment:
Why are we copying that into this class?

Reply:
I originally wrote the code on the basis of having the settings in the `TMap` class - because they are associated with the map strangely enough :wink: - but then I worked out that they would need to be stored as part of the profile and thus as part of the `Host` class. I am aware that some of the members of that class are protected by a `QMutex` but it is not entirely clear why those particular ones are so protected; and rather than having to read the settings from that class for every `T2DMap::paintEvent()` I felt it was safer to work from copies held in the `TMap` class and only access the *master* ones in `Host` when they needed to be changed or used to initialise the copies.

Reply:
You're the one who put a `QMutex` on them... if we don't need it anymore, lets take it off?

Seems to make sense not to complicate things with forgotten mutexes and then copying things around...